components/script/dom/xmlhttprequest.rs

```diff
@@ -979,14 +979,12 @@ impl XMLHttpRequest {
                 // Part of step 11, send() (processing response end of file)
                 // XXXManishearth handle errors, if any (substep 2)
 
-                // Subsubsteps 5-7
+                // Subsubsteps 6-8
                 self.send_flag.set(false);
 
                 self.change_ready_state(XMLHttpRequestState::Done);
                 return_if_fetch_was_terminated!();
-                // Subsubsteps 10-12
-                self.dispatch_response_progress_event(atom!("progress"));
```

Comment:
I've added https://github.com/whatwg/xhr/issues/72#issuecomment-280882420 to clarify what the intention of the spec is here. Assuming that not firing `progress` event when `e.loaded == e.total` is the way forward even in the case of processing a response, the easiest way to fix the issue you raised is to check whether the `ContentLength` is 0 before firing a `progress` event.